app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/compress/RarHelperTask.java

```diff
@@ -82,13 +85,61 @@ void addElements(@NonNull ArrayList<CompressedObjectParcelable> elements)
           elements.add(
               new CompressedObjectParcelable(
                   RarDecompressor.convertName(rarArchive),
-                  0,
-                  rarArchive.getDataSize(),
+                  rarArchive.getMTime().getTime(),
+                  rarArchive.getFullUnpackSize(),
                   rarArchive.isDirectory()));
         }
       }
+    } catch (UnsupportedRarV5Exception e) {
+      throw new ArchiveException("RAR v5 archives are not supported", e);
+    } catch (FileNotFoundException e) {
+      throw new ArchiveException("First part of multipart archive not found", e);
     } catch (RarException | IOException e) {
       throw new ArchiveException(String.format("RAR archive %s is corrupt", fileLocation));
     }
   }
+
+  public static Archive getArchive(String fileLocation) throws RarException, IOException {
```

Comment:
I'd probably move this to `FileUtils`

Reply:
Removed in daae99a since logic to guess first part of RAR is removed.